components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java

```diff
@@ -1404,15 +1404,11 @@ private void storeTokenIdCookie(String sessionId, HttpServletRequest req, HttpSe
         samlssoTokenIdCookie.setSecure(true);
         samlssoTokenIdCookie.setHttpOnly(true);
         samlssoTokenIdCookie.setPath("/");
-        samlssoTokenIdCookie.setMaxAge(defaultMaxAge);
         samlssoTokenIdCookie.setSameSite(SameSiteCookie.NONE);
 
         if (samlssoTokenIdCookieConfig != null) {
-            int age = defaultMaxAge;
-            if (samlssoTokenIdCookieConfig.getMaxAge() > 0) {
-                age = samlssoTokenIdCookieConfig.getMaxAge();
-            }
-            updateSAMLSSOIdCookieConfig(samlssoTokenIdCookie, samlssoTokenIdCookieConfig, age);
+            // -1 arguement was passed in place of null for age as parameter is primitive int.
+            updateSAMLSSOIdCookieConfig(samlssoTokenIdCookie, samlssoTokenIdCookieConfig, -1);
```

Comment:
age -1 means the cookie should expire immediately
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie

Reply:
we could change the int age to integer age in `updateSAMLSSOIdCookieConfig` then pass a null and do a null check in the `updateSAMLSSOIdCookieConfig` for age. this seems a bit better

Reply:
I will look into this. Thanks